App/StackExchange.DataExplorer/Helpers/Captcha.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Net.Http;
+using Newtonsoft.Json;
+
+namespace StackExchange.DataExplorer.Helpers
+{
+    public static class Captcha
+    {
+        public static bool ChallengeIsvalid(NameValueCollection form)
+        {
+            if (IsPassedForCurrentUser())
+            {
+                return true;
+            }
+
+            var challenge = form.Get("g-recaptcha-response", "");
+            bool userIsValidated = false;
+
+            if (challenge.HasValue())
+            {
+                using (var client = new HttpClient())
+                {
+                    var response = client.PostAsync("https://www.google.com/recaptcha/api/siteverify", new FormUrlEncodedContent(new[] {
+                        new KeyValuePair<string, string>("secret", AppSettings.RecaptchaPrivateKey),
+                        new KeyValuePair<string, string>("response", challenge),
+                        new KeyValuePair<string, string>("remoteip", Current.RemoteIP)
+                    })).Result;
```

Comment:
Yep this will deadlock the pool :) I'll yank in and fix the async bits tomorrow ;)

Reply:
Yeah, that would not be ideal! Awesome, thanks! And I'll learn by example :D